sourcery_analytics/extractors.py

```diff
@@ -122,26 +126,30 @@ def from_function(
         """Construct an arbitrary extractor from a function of a node."""
         return cls(FunctionVisitor(function))
 
-    @functools.singledispatchmethod
     def extract(self, item: Extractable) -> typing.Iterator[T]:
         """Extract from source code, a node, a file, or directory."""
+        # `singledispatchmethod` confuses mypy, so wrap with a mypy-friendly interface
+        return self._extract(item)
+
+    @functools.singledispatchmethod
+    def _extract(self, item: Extractable) -> typing.Iterator[T]:
         # Note we use regular dispatch rather than nodedispatch for this function
         # in order to support directories as well as files
         raise NotImplementedError(f"Unable to extract from {item}.")
 
-    @extract.register
+    @_extract.register
     def _extract_from_node(self, node: astroid.nodes.NodeNG) -> typing.Iterator[T]:
         visitor = TreeVisitor[typing.Optional[T], typing.Iterator[typing.Optional[T]]](
             self.visitor
         )
         yield from filter(None, visitor.visit(node))
 
-    @extract.register
+    @_extract.register
     def _extract_from_source(self, source: str) -> typing.Iterator[T]:
-        node = astroid.parse(source)
+        node = self.manager.ast_from_string(clean_source(source))
```

Comment:
**suggestion (docstrings):** Please update the docstring for function: `_extract_from_source`

Reason for update: Change in source parsing method to include error handling.

Suggested new docstring:
```
Extracts nodes from the given source code string using the AstroidManager, with added error handling for syntax errors.
```